react/features/invite/components/InviteButton.web.js

```diff
@@ -0,0 +1,220 @@
+         * Indicates if the "Dial out" feature is available.
+         */
+        _isDialOutAvailable: React.PropTypes.bool,
+
+        /**
+         * The function closing the dialog.
+         */
+        openDialog: React.PropTypes.func,
+
+        /**
+         * Invoked to obtain translated strings.
+         */
+        t: React.PropTypes.func
+    };
+
+    /**
+     * Initializes a new {@code InviteButton} instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        this._onInviteClick = this._onInviteClick.bind(this);
+        this._onInviteOptionSelected = this._onInviteOptionSelected.bind(this);
+        this._updateInviteItems = this._updateInviteItems.bind(this);
+
+        this._updateInviteItems(this.props);
+    }
+
+    /**
+     * Implements React's {@link Component#componentWillReceiveProps()}.
+     *
+     * @inheritdoc
+     * @param {Object} nextProps - The read-only props which this Component will
+     * receive.
+     * @returns {void}
+     */
+    componentWillReceiveProps(nextProps) {
+        if (this.props._isDialOutAvailable !== nextProps._isDialOutAvailable
+            || this.props._isAddToCallAvailable
+                !== nextProps._isAddToCallAvailable) {
+            this._updateInviteItems(nextProps);
+        }
+    }
+
+    /**
+     * Renders the content of this component.
+     *
+     * @returns {ReactElement}
+     */
+    render() {
+        const { t } = this.props;
+
+        const { VERTICAL_FILMSTRIP } = interfaceConfig;
+
+        return (
+            <div className = 'filmstrip__invite'>
+                <div className = 'invite-button-group'>
+                    <Button
+                        onClick = { this._onInviteClick }
+                        shouldFitContainer = { true }>
+                        { t('invite.invitePeople') }
+                    </Button>
+                    <DropdownMenu
+                        items = { this.state.inviteOptions }
+                        onItemActivated = { this._onInviteOptionSelected }
+                        position = { VERTICAL_FILMSTRIP
+                            ? 'bottom right'
+                            : 'top right' }
+                        shouldFlip = { true }
+                        triggerType = 'button' />
+                </div>
+            </div>
+        );
+    }
+
+    /**
+     * Handles the click of the invite button.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onInviteClick() {
+        this.props.openDialog(InviteDialog);
+    }
+
+    /**
+     * Handles selection of the invite options.
+     *
+     * @param { Object } option - The invite option that has been selected from
+     * the dropdown menu.
+     * @private
+     * @returns {void}
+     */
+    _onInviteOptionSelected(option) {
+        this.state.inviteOptions[0].items.map(item => {
```

Comment:
Why are you using map if the returned value isn't returning anything? Is it necessary to even have iteration logic? I'm assuming you can even have the action within the option.

Reply:
I was using it differently before and now it's not needed anymore.